### PR TITLE
Don't lookup mail server when using sendmail

### LINKED ISSUE #22287: Unneccessary SMTP_ADDR lookup when using "PROTOCOL = sendmail"
### Description

I've just upgraded to 1.18.0 and had to fix my [mailer] config to avoid #22274 (and also update my [log] config just to see the problem in the first place!).  Now that I can see warnings in the console I noticed this:

```
2022/12/30 21:49:49 ...es/setting/mailer.go:263:tryResolveAddr() [W] could not look up mailer.SMTP_ADDR: lookup : no such host
```

However it makes no sense for gitea to be attempting to resolve SMTP_ADDR as I'm using the "sendmail" PROTOCOL for my mailer:

```
[mailer]
ENABLED       = true
FROM          = Gitea <gitea@re.da.ct.ed>
PROTOCOL      = sendmail
SENDMAIL_PATH = /usr/sbin/sendmail
SENDMAIL_ARGS = --
```

### Gitea Version

1.18.0

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

Debian 11.6 (amd64)

### How are you running Gitea?

Downloaded from your GitHub release page.

### Database

None

### PR DESCRIPTION
**Pull request by lunny**
Fix #22287

### CODE REVIEW

**Review by yardenshoham — COMMENTED** [review]
If the protocol is `dummy`, will SMTP resolution still occur?

File: modules/setting/mailer.go [hunk 1]
```diff
@@ -176,14 +176,22 @@ func newMailService() {
...
-	if MailService.Protocol == "smtp" {
-		for _, ip := range ips {
-			if !ip.IsLoopback() {
-				log.Warn("connecting over insecure SMTP protocol to non-local address is not recommended")
-				break
+	if MailService.Protocol == "sendmail" {
+		var err error
+		MailService.SendmailArgs, err = shellquote.Split(sec.Key("SENDMAIL_ARGS").String())
+		if err != nil {
+			log.Error("Failed to parse Sendmail args: %s with error %v", CustomConf, err)
```
Comment (delvh) [inline modules/setting/mailer.go]:
```suggestion
			log.Error("Failed to parse Sendmail args: '%s' with error %v", sec.Key("SENDMAIL_ARGS").String(), err)
```

File: modules/setting/mailer.go [hunk 2]
```diff
@@ -176,14 +176,22 @@ func newMailService() {
...
 
-	// we want to warn if users use SMTP on a non-local IP;
-	// we might as well take the opportunity to check that it has an IP at all
-	ips := tryResolveAddr(MailService.SMTPAddr)
-	if MailService.Protocol == "smtp" {
-		for _, ip := range ips {
-			if !ip.IsLoopback() {
-				log.Warn("connecting over insecure SMTP protocol to non-local address is not recommended")
-				break
+	if MailService.Protocol == "sendmail" {
```
Comment (delvh) [inline modules/setting/mailer.go]:
```suggestion
	// we want to warn if users use SMTP on a non-local IP;
	// we might as well take the opportunity to check that it has an IP at all
	// This check is not needed for sendmail
	if MailService.Protocol == "sendmail" {
```

File: modules/setting/mailer.go [hunk 3]
```diff
@@ -176,14 +176,22 @@ func newMailService() {
...
+		var err error
+		MailService.SendmailArgs, err = shellquote.Split(sec.Key("SENDMAIL_ARGS").String())
+		if err != nil {
+			log.Error("Failed to parse Sendmail args: %s with error %v", CustomConf, err)
+		}
+	} else {
+		// we want to warn if users use SMTP on a non-local IP;
+		// we might as well take the opportunity to check that it has an IP at all
```
Comment (delvh) [inline modules/setting/mailer.go]:
```suggestion
```

**Comment by lunny** [comment]
@yardenshoham @delvh done.

**Review by yardenshoham — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**